scripts/clean_test.py

```diff
@@ -0,0 +1,126 @@
+
+        rmtree_swap = self.swap(shutil, 'rmtree', mock_rmtree)
+        exists_swap = self.swap(os.path, 'exists', mock_exists)
+        with rmtree_swap, exists_swap:
+            clean.delete_directory_tree('dir_path')
+        self.assertEqual(check_function_calls, expected_check_function_calls)
+
+    def test_delete_directory_with_existing_dir(self):
+        check_function_calls = {
+            'rmtree_is_called': False
+        }
+        expected_check_function_calls = {
+            'rmtree_is_called': True
+        }
+        def mock_rmtree(unused_path):
+            check_function_calls['rmtree_is_called'] = True
+        def mock_exists(unused_path):
+            return True
+
+        rmtree_swap = self.swap(shutil, 'rmtree', mock_rmtree)
+        exists_swap = self.swap(os.path, 'exists', mock_exists)
+        with rmtree_swap, exists_swap:
+            clean.delete_directory_tree('dir_path')
+        self.assertEqual(check_function_calls, expected_check_function_calls)
+
+    def test_delete_file_with_missing_file(self):
+        check_function_calls = {
+            'remove_is_called': False
+        }
+        expected_check_function_calls = {
+            'remove_is_called': False
+        }
+        def mock_remove(unused_path):
+            check_function_calls['remove_is_called'] = True
+        def mock_isfile(unused_path):
+            return False
+
+        remove_swap = self.swap(os, 'remove', mock_remove)
+        isfile_swap = self.swap(os.path, 'isfile', mock_isfile)
+        with remove_swap, isfile_swap:
+            clean.delete_file('file_path')
+        self.assertEqual(check_function_calls, expected_check_function_calls)
+
+    def test_delete_file_with_existing_file(self):
+        check_function_calls = {
+            'remove_is_called': False
+        }
+        expected_check_function_calls = {
+            'remove_is_called': True
+        }
+        def mock_remove(unused_path):
+            check_function_calls['remove_is_called'] = True
+        def mock_isfile(unused_path):
+            return True
+
+        remove_swap = self.swap(os, 'remove', mock_remove)
+        isfile_swap = self.swap(os.path, 'isfile', mock_isfile)
+        with remove_swap, isfile_swap:
+            clean.delete_file('file_path')
+        self.assertEqual(check_function_calls, expected_check_function_calls)
+
+    def test_function_calls(self):
+        check_function_calls = {
+            'delete_directory_tree_is_called': 0,
+            'delete_file_is_called': 0
+        }
+        expected_check_function_calls = {
+            'delete_directory_tree_is_called': 8,
+            'delete_file_is_called': 4
+        }
+        def mock_delete_dir(unused_path):
+            check_function_calls['delete_directory_tree_is_called'] += 1
+        def mock_delete_file(unused_path):
+            check_function_calls['delete_file_is_called'] += 1
+        def mock_listdir(unused_path):
+            return ['tmpcompiledjs_dir']
```

Comment:
Hmmm, can we control how many files are in this dir? Will this change at any point, and at that time will the test fail?

Reply:
No, we do not need to control to number of files. This directory does not exist. It is just returned to ensure that the main function calls the delete method if any directory is found with this name. I have used a mock for delete as well to ensure that actual delete operation is not performed.